content/en/docs/installation/upgrading/ingress-class-compatibility.md

```diff
@@ -0,0 +1,68 @@
+---
+title: "Notes on Ingress Class Compatibility"
+linkTitle: "Notes on Ingress Class Compatibility"
+weight: 19
+type: "docs"
+---
+
+The following notes apply to anyone upgrading through cert-manager 1.5 or later on Kubernetes v1.19 or later.
+
+# Background
+
+cert-manager 1.5 was released to coincide with Kubernetes 1.22, which
+[removed](https://kubernetes.io/blog/2021/07/14/upcoming-changes-in-kubernetes-1-22/) the `v1beta1`
+Ingress API. As cert-manager creates Ingress resources to solve HTTP-01 challenges, this code path
+needed to be updated.
+
+In the `v1beta1` spec, Ingress Class was a string annotation that was adopted by all popular
+Ingress controllers by convention. In the `v1` spec, `IngressClass` is now its own resource type,
+and the `.spec.ingressClassName` field on `v1` Ingresses is now a reference to that object.
+As the Kubernetes documentation points out, the old and new specs are not directly equivalent.
+
+During the 1.5 and 1.6 cert-manager release cycles, we discovered that ingress controllers have
+handled the graduation of Ingress to `v1` differently. Some treat the class as an opaque string,
+similarly to the annotation. Some were unintentionally broken, as their default ingress class name
+contains characters that are disallowed in object references, e.g. (`/`). Some now require you to
+create an `IngressClass` object matching the field to work.
+
+cert-manager aims to be compatible with as many ingress controllers as possible. According to the
+Ingress v1 [Kubernetes enhancement proposal], the deprecated annotation, if present, takes
+precedence over the new field. From our perspective, the option that maintains the highest
+compatibility is to only use the annotation, even when creating `v1` Ingresses
+
+[Kubernetes enhancement proposal]: https://github.com/kubernetes/enhancements/tree/44dd2975dc6cdad96ca73e7b0ba1794f1196f604/keps/sig-network/1453-ingress-api#interoperability-with-previous-annotation
+
+# Notes For Specific Ingress Controllers
```

Comment:
People reading this page in a "standalone" way (without reading the 1.7 release notes first) won't know whether they should read the rest of the document or not. I think we should explain if they are affected or not by adding which cert-manager versions are affected.

Reply:
https://github.com/cert-manager/website/pull/812/files#diff-0253be5163bbb956f00cbf9d08ea6281c91d9b75ea5cd83ff331ff0c256d6923R8 

It is at the top of the page, but it may not be good enough. Do you have a suggestion?

Reply:
```suggestion
# Notes For Specific Ingress Controllers

The regression is only present in cert-manager 1.5.0, 1.5.1, 1.5.2, 1.5.3, 1.5.4, 1.6.0, 1.6.1, 1.6.2, and 1.6.3, and applies to clusters running Kubernetes 1.19+ when using an Issuer or ClusterIssuer with an ACME HTTP-01 solver configured. 

If you are upgrading from one of the above versions to 1.5.5+, 1.6.4+ or 1.7.0+, and that you are in the following case:

- You are using Gloo, Contour, Skipper, or kube-ingress-aws-controller;
- You are using Traefik, Istio, Ambassador, or ingress-nginx with their default "class" (e.g., `istio` for Istio);

then upgrading from a version that contains the regression to 1.7.0, 1.6.4 or 1.5.5 won't have any impact and you can skip the rest of this document.

If you are using an ingress controller that doesn't belong to the above, upgrading from a version that contains the regression to 1.7.0, 1.6.4 or 1.5.5 may have an impact. The rest of this document details the measures you will need to take.

```

Reply:
```suggestion
# Notes For Specific Ingress Controllers

When upgrading from one of the affected versions of cert-manager (1.5.0, 1.5.1, 1.5.2, 1.5.3, 1.5.4, 1.6.0, 1.6.1, 1.6.2, and 1.6.3) to 1.5.5, 1.6.4 or 1.7.0 and above, and that you are in one or multiple of the following cases:

- You are using Gloo, Contour, Skipper, or kube-ingress-aws-controller,
- You are using Traefik, Istio, Ambassador, or ingress-nginx with their default "class" (e.g., `istio` for Istio),

then you don't need to worry and the upgrade will have no impact. You can skip the rest of this document.

If you are using an ingress controller that doesn't belong to the above, upgrading from a version that contains the regression to 1.5.5, 1.6.4 or 1.7.0 and above may have an impact. The rest of this document details the measures you will need to take.

```

Reply:
I am OK for moving this suggestion to a new PR.